modules/ROOT/pages/customer-connector-upgrade.adoc

```diff
@@ -0,0 +1,162 @@
+= Customer Guide for Upgrading and Testing Custom Connectors for Java 17
+
+Upgrade and test your custom connectors for Java 17 to ensure compatibility within the MuleSoft ecosystem. 
+
+== Before You Begin
+
+Before upgrading and testing your custom connector for Java 17, you must be familiar with the following:
+
+* *How to run and compile your custom connector*
++
+Ensure your custom connector runs with Java 17, but always use Java 8 as the target compilation level.
++
+Because you must compile your custom connector with Java 8, use the Java compiler configuration from the Java SDK parent POM. You can’t use language features from Java 11 or Java 17. 
+
+* *-add-opens or -add-export directives*
++
+Do not set `-add-opens` or `-add-export` directives.
+
+* *Minimum Mule version caveats*
++
+If your custom connector is running on Mule 4.5.x or earlier, you can’t run Java 17 tests that extend the MuleArtifactFunctionalTestCase, but you can still run Java 8 tests.
++
+To decouple the minimum Mule version when running tests, add the `-Dmunit.jvm` parameter to use Java 17 as described in <<test-your-custom-connector-with-munit>>.
+
+[[upgrade-your-mule-sdk-connectors]]
+== Upgrade Your Custom Mule SDK Connectors
```

Comment:
I think this should be a gerund, if all the H3s are subtasks.

Reply:
So in this case, H1s, H2s, and H3s are gerunds?

Reply:
I think the H1 and H2 are gerunds and the H3 are imperative, since they are the tasks that contain the steps.